File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/util/OptionsBean.java
package org.akoshterek.backgammon.util;

/**
 * @author Alex
 *         date 04.08.2015.
 */
public class OptionsBean {
    private boolean isHelp;
    private boolean isWarranty;
    private boolean isLicense;
    private String[] agentNames;
    private String benchmarkAgentName;
    private int trainingGames;
    private int benchmarkGames;
    private int benchmarkPeriod;
    private boolean isVerbose;

    public boolean isHelp() {
        return isHelp;
    }

    public void setIsHelp(boolean isHelp) {
        this.isHelp = isHelp;
    }

    public boolean isWarranty() {
        return isWarranty;
    }

    public void setIsWarranty(boolean isWarranty) {
        this.isWarranty = isWarranty;
    }

    public boolean isLicense() {
        return isLicense;
    }

    public void setIsLicense(boolean isLicense) {
        this.isLicense = isLicense;
    }

    public String[] getAgentNames() {
        return agentNames;
    }

    public void setAgentNames(String[] agentNames) {
        this.agentNames = agentNames;
    }

    public String getBenchmarkAgentName() {
        return benchmarkAgentName;
    }

    public void setBenchmarkAgentName(String benchmarkAgentName) {
        this.benchmarkAgentName = benchmarkAgentName;
    }

    public int getTrainingGames() {
        return trainingGames;
    }

    public void setTrainingGames(int trainingGames) {
        this.trainingGames = trainingGames;
    }

    public int getBenchmarkGames() {
        return benchmarkGames;
    }

    public void setBenchmarkGames(int benchmarkGames) {
        this.benchmarkGames = benchmarkGames;
    }

    public int getBenchmarkPeriod() {
        return benchmarkPeriod;
    }

    public void setBenchmarkPeriod(int benchmarkPeriod) {
        this.benchmarkPeriod = benchmarkPeriod;
    }

    public boolean isVerbose() {
        return isVerbose;
    }

    public void setIsVerbose(boolean isVerbose) {
        this.isVerbose = isVerbose;
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/util/OptionsBean.scala
package org.akoshterek.backgammon.util

class OptionsBean {
  var isHelp: Boolean = false
  var isWarranty: Boolean = false
  var isLicense: Boolean = false
  private var _agentNames: Vector[String] = Vector()
  var benchmarkAgentName: String = ""
  var trainingGames: Int = 0
  var benchmarkGames: Int = 0
  var benchmarkPeriod: Int = 0
  var isVerbose: Boolean = false
  var alpha: Float = 0.01f
  var lambda: Float = 0.7f
  var gamma: Float = 1.0f
  var experimentPath: String = ""
  var experimentRunTag: String = ""
  var trainingOpponents: String = "self:100"
  var benchmarkOpponents: String = "SimpleHeuristic"
  var useOutputBias: Boolean = true

  var gaTraining: Boolean = false
  var gaPopulationSize: Int = 20
  var gaGenerations: Int = 10
  var gaEliteCount: Int = 4
  var gaMutationRate: Float = 0.05f
  var gaMutationStrength: Float = 0.1f

  def agentNames: Vector[String] = _agentNames
  def agentNames_= (agents: Array[String]): Unit = {
    agents match {
      case null => _agentNames = Vector()
      case _ => _agentNames = agents.toVector
    }
  }
}